packages/@internationalized/date/src/queries.ts

```diff
@@ -225,17 +242,21 @@ function getRegion(locale: string): string | undefined {
   return part === 'u' ? undefined : part;
 }
 
-function getWeekStart(locale: string): number {
+export function getWeekStart(locale: string): number {
   // TODO: use Intl.Locale for this once browsers support the weekInfo property
   // https://github.com/tc39/proposal-intl-locale-info
   let region = getRegion(locale);
   return region ? weekStartData[region] || 0 : 0;
 }
 
 /** Returns the number of weeks in the given month and locale. */
-export function getWeeksInMonth(date: DateValue, locale: string): number {
-  let days = date.calendar.getDaysInMonth(date);
-  return Math.ceil((getDayOfWeek(startOfMonth(date), locale) + days) / 7);
+export function getWeeksInMonth(date: DateValue, locale: string, firstDayOfWeek?: DayOfWeek): number {
+  let monthStart = startOfMonth(date);
+  let firstVisibleDay = startOfWeek(monthStart, locale, firstDayOfWeek);
+  let monthEnd = endOfMonth(date);
+  let lastVisibleDay = startOfWeek(monthEnd, locale, firstDayOfWeek).add({days: 6});
```

Comment:
Mega nit: 
No need to do this in this PR, but if we add `firstDayOfWeek` to `endOfWeek`, this could be modified to call that function instead since they both do the same `.add` call (alternatively, you could just call `.add` to `monthStart` since `.add` creates a new date time). Not sure how useful that would be but would give parity between those two functions.